### PR TITLE
Add ReamazeAPI

### PR DESCRIPTION
**Pull request by itspriddle**

### CODE REVIEW

File: lib/reamaze_api/client.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+module ReamazeAPI
+  class Client
+    class Middleware < FaradayMiddleware::ParseJson
+      define_parser do |body|
+        begin
+          JSON.parse(body) unless body.strip.empty?
+        rescue
+          body
+          # Utils.error_hash(e).merge(payload: body)
```
Comment (fabiokr) [inline lib/reamaze_api/client.rb]: Commented code

Comment (rubemz) [inline lib/reamaze_api/client.rb]: remove comment?

File: lib/reamaze_api/client.rb [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      end
+    end
+
+    Faraday::Response.register_middleware reamaze_api: lambda { Middleware }
+
+    # Public: HTTP methods used by the API
+    HTTP_METHODS = %i(get put post)
+
+    # Public: URL for Reamaze API.
+    API_URL = "https://%{brand}.reamaze.com/api/v1".freeze
```
Comment (fabiokr) [inline lib/reamaze_api/client.rb:28]: Do they have a test env?

File: lib/reamaze_api/client.rb [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      @url = URI.parse(API_URL % { brand: brand })
+
+      @http = Faraday.new(url: @url, ssl: { verify: true }) do |builder|
+        builder.request    :json
+        builder.response   :json
+        builder.adapter    Faraday.default_adapter
+        builder.basic_auth login, token
+      end
+
+      @articles      = Article.new(self)
```
Comment (fabiokr) [inline lib/reamaze_api/client.rb]: Maybe make these lazy loaded?

File: test/support/assert_api_response.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    end
+  end
+end
+
+Minitest::Expectation.class_eval do
+  %i(to_get to_put to_post).each do |method|
+    define_method method do |*args|
+      ctx.assert_api_request method[3..-1], *args do |client|
+        target.call client if Proc === target
+        # client.instance_eval(&target) if Proc === target
```
Comment (fabiokr) [inline test/support/assert_api_response.rb]: Comment

File: lib/reamaze_api/utils/hash_keys.rb [hunk 5]
```diff
@@ -0,0 +1,182 @@
+module ReamazeAPI
+  module Utils
+    module HashKeys
+      refine ::Hash do
```
Comment (rubemz) [inline lib/reamaze_api/utils/hash_keys.rb:4]: :+1:

File: test/support/assert_api_response.rb (hunk 4 above)
Comment (rubemz) [inline test/support/assert_api_response.rb]: comment

File: lib/reamaze_api/client.rb (hunk 2 above)
Comment (itspriddle) [inline lib/reamaze_api/client.rb:28]: Not that I saw :(

**Comment by newx** [comment]
:+1:

**Pull request merged**